### PR TITLE
Reconsent flow: Disease preferences screen amends following QA

### PR DESCRIPTION
**Pull request by juliatan**
# Description

Covers tickets:
https://www.notion.so/joinzoe/2021-07-08-Reconsent-Flow-f611979df8794da288763d939115ecc1?p=70ab59cf2bab4e709f14de601259a4d5

https://www.notion.so/joinzoe/2021-07-08-Reconsent-Flow-f611979df8794da288763d939115ecc1?p=6d7e162e8acb44189a7feb39637d999e

https://www.notion.so/joinzoe/Fix-screen-2-disease-cards-see-Sol-s-comments-on-designs-above-Also-fix-mental-health-copy-246701d1a4624eb686fd4681ac025290

https://www.notion.so/joinzoe/Disease-cards-on-back-click-definitely-need-to-save-to-Redux-state-Think-about-Android-back-clic-d38f03dc99734d9690962615b98763c6
Question for @aikewoody : What did you mean by Android not having a back click listener for this ticket? Also, will my amend to the back button onPress event affect the e2e tests?

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Simulator Screen Shot - iPhone 11 - 2021-07-13 at 19 25 25

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

**Comment by aikewoody** [comment]
Looks good, only thing is we need to update the state when on Android the user navigates back with the hardware / software back button: https://reactnative.dev/docs/backhandler

**Review by aikewoody — CHANGES_REQUESTED** [review]
Looks good, only thing is we also need to update the state when on Android the user navigates back with the hardware / software back button: https://reactnative.dev/docs/backhandler

File: assets/lang/en.json [hunk 1]
```diff
@@ -1736,11 +1736,11 @@
     },
     "research_consent_mental_health": {
       "name": "Mental Health",
-      "description": "Up to 10% of people in the UK will experience depression"
+      "description": "Up to 20% of people in the UK will experience depression"
```
Comment (bionara) [inline assets/lang/en.json:1739]: it doubled!

**Review by bionara — APPROVED** [review]

**Comment by juliatan** [comment]
Added Android hardware back button handler.
Jul-14-2021 11-51-36

File: src/features/reconsent/screens/ReconsentDiseasePreferencesScreen.tsx [hunk 2]
```diff
@@ -41,6 +43,23 @@ export default function ReconsentDiseasePreferencesScreen() {
...
+    dispatch(updateDiseasePreferences(diseasePreferences));
+  };
+
+  const androidBackHandler = () => {
+    updateState();
+    navigation.goBack();
+    return true;
+  };
+
+  React.useEffect(() => {
```
Comment (bionara) [inline src/features/reconsent/screens/ReconsentDiseasePreferencesScreen.tsx:56]: Very nice!

**Review by bionara — APPROVED** [review]

**Pull request merged**